app/bundles/CoreBundle/Helper/CoreParametersHelper.php

```diff
@@ -71,4 +73,28 @@ public function hasParameter($name)
     {
         return $this->parameterBag->has('mautic.'.$name);
     }
+
+    /**
+     * @return int
+     */
+    public function getMaxUploadSize()
+    {
+        return $this->getParameter('max_size');
+    }
+
+    /**
+     * @return array
+     */
+    public function getAllowedExtensionsForUpload()
+    {
+        return $this->getParameter('allowed_extensions');
+    }
+
+    /**
+     * @return string
+     */
+    public function getUploadDirForForms()
+    {
+        return $this->getParameter(self::PARAMETERS_FORM_UPLOAD_DIR);
+    }
```

Comment:
I'm not sure about adding in these proxy methods. If we do, to be consistent, we need to add one for every possible parameter. Otherwise we end up using the `CoreParametersHelper` in different ways depending solely on the key that we are getting.

Reply:
Problem with previous approach is that strings as `form_upload_dir` are used in every place which need this parameter. There is no check for typo.
Second thing is that this hides implementation details and we are free to change it later.
That is why I think this is good approach and should be used for every new parameter (and for old ones after refactoring).

Reply:
This would be ok if we only had to worry about our core code. But the CoreParameterHelper has to support all parameters configured by plugins. This means we can't hard code the parameter names unless we provide some kind of API for plugins.

Reply:
Getters removed, related changes reverted.